### PR TITLE
remove usage of net exe

### PR DESCRIPTION
**Pull request by arjgupta**

### CODE REVIEW

File: Tasks/WindowsMachineFileCopy/RoboCopyJob.ps1 [hunk 1]
```diff
@@ -233,54 +220,30 @@ param (
...
-        if ($LASTEXITCODE -ne 0) 
-        {
-            $errorMessage = (Get-VstsLocString -Key "WFC_FailedToConnectToPathWithUser" -ArgumentList $machineShare, $($credential.UserName)) + $dtl_mapOut
-            ThrowError -errorMessage $errorMessage -fqdn $fqdn
-        }
+        New-PSDrive -Name "WFCPSDrive" -PSProvider FileSystem -Root $destinationNetworkPath -Credential $psCredentialObject -ErrorAction 'Stop'
     }
-
-    try
+
```
Comment (chshrikh) [inline Tasks/WindowsMachineFileCopy/RoboCopyJob.ps1]: use try catch and get the logs in exception case, will help us to debug if required

Comment (arjgupta) [reply Tasks/WindowsMachineFileCopy/RoboCopyJob.ps1]: yes, that sounds right. I'll fix it.

File: Tasks/WindowsMachineFileCopy/task.loc.json [hunk 2]
```diff
@@ -12,8 +12,8 @@
   "author": "Microsoft Corporation",
   "version": {
     "Major": 2,
-    "Minor": 0,
-    "Patch": 4
+    "Minor": 1,
```
Comment (chshrikh) [inline Tasks/WindowsMachineFileCopy/task.loc.json]: just update patch version only, if we are not sure what is behavior with updating the Minor version

Comment (chshrikh) [reply Tasks/WindowsMachineFileCopy/task.loc.json]: lets check with @kmkumaran

File: Tasks/WindowsMachineFileCopy/RoboCopyJob.ps1 [hunk 3]
```diff
@@ -22,6 +22,7 @@ param (
 
     $sourcePath = $sourcePath.Trim().TrimEnd('\', '/')
     $targetPath = $targetPath.Trim().TrimEnd('\', '/')    
+    $psCredentialObject = New-Object PSCredential($credential.UserName, (ConvertTo-SecureString $credential.Password -AsPlainText -Force))
```
Comment (chshrikh) [inline Tasks/WindowsMachineFileCopy/RoboCopyJob.ps1]: we can remove the old network credentials if not used any more in this file

File: Tasks/WindowsMachineFileCopy/RoboCopyJob.ps1 [hunk 4]
```diff
@@ -233,54 +220,30 @@ param (
...
-        }
-        $command += " 2>&1"
-        
-        $dtl_mapOut = iex $command
-        if ($LASTEXITCODE -ne 0) 
-        {
-            $errorMessage = (Get-VstsLocString -Key "WFC_FailedToConnectToPathWithUser" -ArgumentList $machineShare, $($credential.UserName)) + $dtl_mapOut
-            ThrowError -errorMessage $errorMessage -fqdn $fqdn
-        }
+        New-PSDrive -Name "WFCPSDrive" -PSProvider FileSystem -Root $destinationNetworkPath -Credential $psCredentialObject -ErrorAction 'Stop'
```
Comment (chshrikh) [inline Tasks/WindowsMachineFileCopy/RoboCopyJob.ps1]: test once with Server 2008 r2 and server 2012

File: Tasks/WindowsMachineFileCopy/RoboCopyJob.ps1 [hunk 5]
```diff
@@ -218,69 +202,55 @@ param (
...
-    $userName = Get-DownLevelLogonName -fqdn $fqdn -userName $($credential.UserName)
-    $password = $($credential.Password) 
 
-    if([bool]([uri]$targetPath).IsUnc)
-    {
-        Create-DestinationDirectory -path $destinationNetworkPath
-    }
-    
-    $netExeCommand = Get-NetExeCommand
+    Create-DestinationDirectory -path $destinationNetworkPath
```
Comment (rajatagrawal-dev) [inline Tasks/WindowsMachineFileCopy/RoboCopyJob.ps1]: This method was only tested for UNC target paths. Is removing the UNC target path check necessary?

Comment (arjgupta) [reply Tasks/WindowsMachineFileCopy/RoboCopyJob.ps1]: destinationNetworkPath is always a unc path

**Pull request merged**